### PR TITLE
Add GitHub attribute model

### LINKED ISSUE #42: Add GitHub class to Model.
Add the GitHub class to be used as a field for the Person Model.

### PR DESCRIPTION
**Pull request by bryans17**
1. Add new github attribute field for Person Model.
2. Updated AddCommand and EditCommand to work with the new github attribute field.
3. Updated test code to work with the new github attribute field.
4. Updated test json files to work with the new github attribute field.

Fixes #42.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -135,10 +135,10 @@ Format: `labGrade i/INDEX l/LAB m/MARK`
 
 Action | Format, Examples
 --------|------------------
-**Add** | `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+**Add** | `add n/NAME g/GITHUB t/TELEGRAM_HANDLE s/StudentID` <br> e.g., `add n/James Ho g/jamesH t/jamesho s/A0123456T`
 **Clear** | `clear`
 **Delete** | `delete INDEX`<br> e.g., `delete 3`
-**Edit** | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
+**Edit** | `edit INDEX [n/NAME] [g/GITHUB] [t/TELEGRAM_HANDLE] [s/StudentID]`<br> e.g.,`edit 2 n/James Lee g/jamesHo`
```
Comment (geetengtan) [inline docs/UserGuide.md:141]: I assume this is the end result? we going to remove the phone, email and address at the end, but for now, it's only changed here?

File: src/main/java/seedu/address/model/person/GithubUsername.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+public class GithubUsername {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Github usernames should only contain at most 39 alphanumeric characters and hyphens,"
+                    + " but cannot have consecutive hyphens and cannot begin or end with hyphens.";
+
+    /*
+     * The first and last character of the GitHub Username must a hyphen,
+     * can't have consecutive hyphens in the GitHub Username,
+     * and can't be longer than 39 alphanumeric characters.
```
Comment (geetengtan) [inline src/main/java/seedu/address/model/person/GithubUsername.java:20]: * The first and last character of the GitHub Username must not be a hyphen,
* can't have consecutive hyphens in the GitHub Username,
* and can't be longer than 39 alphanumeric characters.

**Review by geetengtan — COMMENTED** [review]
Small issue on the comment

File: docs/UserGuide.md (hunk 1 above)
Comment (bryans17) [reply docs/UserGuide.md:141]: Yes it's the end result. Maybe we shouldn't change it at this stage. I will revert it back to the old version.

File: src/main/java/seedu/address/model/person/GithubUsername.java (hunk 2 above)
Comment (bryans17) [reply src/main/java/seedu/address/model/person/GithubUsername.java:20]: Thanks for the catch. I will update it.

File: docs/UserGuide.md (hunk 1 above)
Comment (bryans17) [reply docs/UserGuide.md:141]: Sorry, just realized that our feature list already included the end result so I think following it in this section of the user guide is appropriate. Will leave it in.

**Review by geetengtan — APPROVED** [review]

**Pull request merged**